### PR TITLE
Unify pushSubscription/publicVapidKey retrievals

### PR DESCRIPTION
**Pull request by pinarx**
cc @gauntface

Should be a no-op in terms of functionality:
1) Use async/await in the ControllerInterface.
2) Merge the retrieval of pushSubscription and publicVapidKey, so we don't call getPushSubscription and getPublicVapidKey multiple times in the flow.

### CODE REVIEW

File: packages/messaging/src/controllers/controller-interface.ts [hunk 1]
```diff
@@ -100,26 +108,31 @@ export default class ControllerInterface {
...
+          swReg,
+          pushSubscription,
+          publicVapidKey,
+          tokenDetails
+        );
       }
-    });
+    }
+
+    // If the VAPID details are updated, delete the existing token,
```
Comment (gauntface) [inline packages/messaging/src/controllers/controller-interface.ts]: May reword this to as the logic suggests the delete and create steps are undertaken if the token is not longer, like when VAPID Details are changed.

Comment (pinarx) [reply packages/messaging/src/controllers/controller-interface.ts]: done
